src/components/trail.component.js

```diff
@@ -3,68 +3,52 @@
 import React, {
   Component,
   View,
+  Text,
   StyleSheet,
   MapView,
   Image,
   Navigator
 } from 'react-native';
-import Firebase from 'firebase';
-import X2JS from 'x2js';
-import FairLandXML from '../gpx/fairland';
 import s from './trail.style';
 const styles = StyleSheet.create(s);
 
 export default class TrailMap extends Component {
 
-  constructor(props) {
-
-    super(props);
-
-    this.state = {
-      isFirstLoad: true,
-      mapRegion: undefined,
-      annotations: []
-    };
-  }
-
   componentWillMount() {
-    //Firebase.enableLogging(true);
-    this.ref = new Firebase('https://shining-fire-7029.firebaseio.com/annotations');
-    this.ref.push({
-      latitude: '42.086445',
-      longitude: '-76.918551',
-      title: 'test1title',
-      subtitle: 'subtitle1test'
-    });
+    this.props.fetchAnnotations();
+    this.props.fetchTrails();
   }
 
   render() {
     return (
       <Navigator
         renderScene={this.renderScene}
-      />
+        />
     );
   }
 
-  renderScene = (route, navigator) => {
-    const x2js = new X2JS();
-    const fairland = x2js.xml2js(FairLandXML);
-    const mapCoords = fairland.gpx.trk[0].trkseg.trkpt.map((o)=> {
-      return {
-        latitude: parseFloat(o._lat),
-        longitude: parseFloat(o._lon)
-      };
-    });
+  renderScene = () => {
+
+    const { trails, isFetching } = this.props.trails;
+
+    if( trails.length === 0 || isFetching ){
+      console.log('trails are empty', trails, isFetching);
+      return (
+        <View style={styles.container}>
+          <MapView style={styles.map}/>
```

Comment:
View will initially render on the entire US. Once the redux store returns map data, it will zoom in on the map. Nice effect :)